include/oneapi/dpl/experimental/kt/two_pass_scan.h

```diff
@@ -149,15 +149,115 @@ sub_group_scan_partial(const SubGroup& sub_group, ValueType& value, BinaryOp bin
     }
 }
 
+template <std::uint8_t VL, bool IsInclusive, bool InitPresent, bool CaptureOutput, std::size_t J_max, typename SubGroup,
+          typename UnaryOp, typename BinaryOp, typename LazyValueType, typename InRange, typename OutRange>
+void
+scan_through_elements_helper(const SubGroup& sub_group, UnaryOp unary_op, BinaryOp binary_op,
+                             LazyValueType& sub_group_carry, InRange __in_rng, OutRange __out_rng,
+                             std::size_t start_idx, std::size_t M, std::size_t J, std::size_t subgroup_start_idx,
+                             std::size_t sub_group_id, std::size_t active_subgroups)
```

Comment:
`sub_group_id` can be `std::uint32_t` as that is the return type from the query in the SYCL spec.

It may be worth looking into if using smaller sized types shows any benefit for some of these variables where it is applicable but that is larger task beyond here this PR.

Reply:
Its true, we can probably shrink a lot of these types down.  I can try to see what low hanging fruit there is here.

Reply:
I made some improvements to the types.